larsim/LegacyLArG4/ParticleListAction.cxx

```diff
@@ -214,15 +217,15 @@ namespace larg4 {
       // if not, then see if it is possible to walk up the fParentIDMap to find the
       // ultimate parent of this particle.  Use that ID as the parent ID for this
       // particle
-      if (!fparticleList->KnownParticle(parentID)) {
+      if (!fparticleList->KnownParticle(parentID) && (!fdroppedParticleList || !fdroppedParticleList->KnownParticle(parentID))) {
         // do add the particle to the parent id map
         // just in case it makes a daughter that we have to track as well
         fParentIDMap[trackID] = parentID;
         int pid = this->GetParentage(parentID);
 
         // if we still can't find the parent in the particle navigator,
         // we have to give up
-        if (!fparticleList->KnownParticle(pid)) {
+        if (!fparticleList->KnownParticle(pid) && (!fdroppedParticleList || !fdroppedParticleList->KnownParticle(pid))) {
```

Comment:
For clarity, I suggest re-expressing the logic without logical ORs:

```c++
if (!fparticleList->KnownParticle(pid) && !(fdroppedParticleList && fdroppedParticleList->KnownParticle(pid)))
```

Reply:
Done!